openverse_catalog/dags/common/dag_factory.py

```diff
@@ -87,20 +149,71 @@ def create_provider_api_workflow(
-                depends_on_past=False,
-            )
+            pull_kwargs["args"] = [DATE_RANGE_ARG_TEMPLATE % day_shift]
+
+        pull_data = PythonOperator(
+            task_id=f"pull_{media_type_name}_data",
+            python_callable=_push_output_paths_wrapper,
+            op_kwargs=pull_kwargs,
+            depends_on_past=False,
+            execution_timeout=dagrun_timeout if dated else None,
+        )
+
+        for media_type in media_types:
+            with TaskGroup(group_id=f"load_{media_type}_data") as load_data:
+                create_loading_table = PythonOperator(
+                    task_id="create_loading_table",
+                    python_callable=sql.create_loading_table,
+                    op_kwargs={
+                        "postgres_conn_id": DB_CONN_ID,
+                        "identifier": identifier,
+                        "media_type": media_type,
+                    },
+                    trigger_rule=TriggerRule.NONE_SKIPPED,
+                    doc_md="Create a temporary loading table for "
+                    f"ingesting {media_type} data from a TSV",
+                )
+                copy_to_s3 = PythonOperator(
+                    task_id="copy_to_s3",
+                    python_callable=s3.copy_file_to_s3,
+                    op_kwargs={
+                        "tsv_file_path": XCOM_PULL_TEMPLATE
+                        % (pull_data.task_id, f"{media_type}_tsv"),
+                        "s3_bucket": OPENVERSE_BUCKET,
+                        "s3_prefix": f"{media_type}/{provider_name}",
+                        "aws_conn_id": AWS_CONN_ID,
+                    },
+                    trigger_rule=TriggerRule.NONE_SKIPPED,
+                )
+                load_from_s3 = PythonOperator(
+                    task_id="load_from_s3",
+                    python_callable=loader.load_from_s3,
+                    op_kwargs={
+                        "bucket": OPENVERSE_BUCKET,
+                        "key": XCOM_PULL_TEMPLATE % (copy_to_s3.task_id, "s3_key"),
+                        "postgres_conn_id": DB_CONN_ID,
+                        "media_type": media_type,
+                        "tsv_version": XCOM_PULL_TEMPLATE
+                        % (copy_to_s3.task_id, "tsv_version"),
+                        "identifier": identifier,
+                    },
+                )
+                drop_loading_table = PythonOperator(
+                    task_id="drop_loading_table",
+                    python_callable=sql.drop_load_table,
+                    op_kwargs={
+                        "postgres_conn_id": DB_CONN_ID,
+                        "identifier": identifier,
+                        "media_type": media_type,
+                    },
+                    trigger_rule=TriggerRule.ALL_DONE,
+                )
+                [create_loading_table, copy_to_s3] >> load_from_s3
+                [create_loading_table, load_from_s3] >> drop_loading_table
```

Comment:
What's going on here? :magic_wand:

Reply:
I think this is [setting the dependencies](https://airflow.apache.org/docs/apache-airflow/stable/concepts/tasks.html#relationships) for the tasks within the `load_data` TaskGroup. 

@AetherUnbound I'm curious if this is logically equivalent to:
```
[create_loading_table, copy_to_s3] >> load_from_s3 >> drop_loading_table
```
? I think it's still clear this way just wondering what the difference is.

I understand the sequence here but it took a struggle, mostly because of unfamiliarity with Airflow (and the actual process!). I think it would be worth adding some comments (or maybe expanding the docstring for the factory method) to describe the flow.

Reply:
:exploding_head: 

This is cryptic, but I understand that if you know Airflow already then it's pretty normal syntax! I'm happy to just accept it as is (and learn more about it at a later date!) :slightly_smiling_face:

Reply:
Oh that's perfect! Definitely, I think copying it over would make a lot of sense 😄